### PR TITLE
Feat/dynamic page number llm whisperer

### PR DESCRIPTION
**Pull request by kirtimanmishrazipstack**
## What

... Add `page-separator` field in llm-whisperer

## Why

... Need to have this field in llm-whisperer

## How

... Changes on llm-whisperer adapter

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing

1. Editable install on `unstract/sdk` on backend
2. Tested pdf raw view with this `unstract/sdk` changes
3. pdf format is <<< 1 >>>, if page_seperator format is <<< {{page_no}} >>>
4. pdf format is <<< >>>,  if page_seperator format is empty

## Screenshots

...
Screenshot from 2024-08-06 12-25-49
Screenshot from 2024-08-06 12-24-22

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json [hunk 1]
```diff
@@ -83,6 +83,12 @@
       "default": "",
       "pattern": "^(\\s*\\d+-\\d+|\\s*\\d+-|\\s*\\d+|^$)(,\\d+-\\d+|,\\d+-|,\\d+)*$",
       "description": "Specify the range of pages to extract (e.g., 1-5, 7, 10-12, 50-). Leave it empty to extract all pages."
+    },
+    "page_seperator": {
+      "type": "string",
+      "title": "Page separator",
+      "default": "<<< >>>",
+      "description": "Specify how the pages to separate (e.g. << {{page_no}} >>, << >>). Leave the {{page_no}} empty if you don't want the page number as page separator."
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json:91]: Can change to something like:
Specify a pattern to separate the pages in the document. ( e.g. << {{page_no}} >>, << >>). This pattern will be inserted at the end of every page. Leave the {{page_no}} empty if you don't want the page number as page separator."

Comment (Deepak-Kesavan) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json:91]:
```suggestion
      "description": "Specify a pattern to separate the pages in the document (e.g., << {{page_no}} >>, << >>). This pattern will be inserted at the end of every page. Omit {{page_no}} if you don't want to include the page number in the separator."
```

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json [hunk 2]
```diff
@@ -83,6 +83,12 @@
       "default": "",
       "pattern": "^(\\s*\\d+-\\d+|\\s*\\d+-|\\s*\\d+|^$)(,\\d+-\\d+|,\\d+-|,\\d+)*$",
       "description": "Specify the range of pages to extract (e.g., 1-5, 7, 10-12, 50-). Leave it empty to extract all pages."
+    },
+    "page_seperator": {
```
Comment (Deepak-Kesavan) [inline src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json:87]: @kirtimanmishrazipstack Are we not adding the regex pattern?

Comment (kirtimanmishrazipstack) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json:87]: @Deepak-Kesavan had a discussion with @arun-venkataswamy , we won't be going with regex for `page_separator`.  In code we will just find and replace `{{page_no}}`

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json (hunk 1 above)
Comment (kirtimanmishrazipstack) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json]: @Deepak-Kesavan done

**Review by Deepak-Kesavan — APPROVED** [review]

**Review by gaya3-zipstack — APPROVED** [review]
Looks fine.

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json (hunk 1 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/static/json_schema.json]: @kirtimanmishrazipstack Version needs to be updated to next minor based on when it gets in.

**Review by vishnuszipstack — APPROVED** [review]

**Review by tahierhussain — APPROVED** [review]

**Review by vishnuszipstack — APPROVED** [review]

**Pull request merged**